samcli/config/config.py

```diff
@@ -0,0 +1,233 @@
+
+        Returns
+        -------
+        Boolean
+        """
+
+        LOG.debug("Validating SAMRC config with given JSONSchema")
+        jsonschema.validate(config, schema)
+
+        LOG.debug("SAMRC looks valid!")
+        LOG.debug("Schema used: %s", schema)
+        LOG.debug("Config used: %s", config)
+
+        return True
+
+    def __find_config(self):
+        """Looks up for user and project level config
+
+        Returns
+        -------
+        Tuple
+            Tuple with both configs and whether they were found
+
+        Example
+        -------
+            > user_config, project_config = self.__find_config()
+        """
+
+        self.__user_config_file = Path.home().joinpath('.samrc')
+        self.__project_config_file = Path.cwd().joinpath('.samrc')
+
+        if self.__has_user_config() and self.__has_project_config():
+            self.config_file = (self.__user_config_file,
+                                self.__project_config_file)
+        elif self.__has_project_config():
+            self.config_file = None, self.__project_config_file
+        elif self.__has_user_config():
+            self.config_file = self.__user_config_file, None
+        else:
+            self.config_file = None, None
+
+        return self.config_file
+
+    def merge_config(self, user_config, project_config):
+        """Merge project and user configuration into a single dictionary
+
+        Creates a new configuration with both configuration merged
+        it favours project level over user configuration if keys are duplicated
+
+        NOTE
+        ----
+            It takes any number of nested dicts
+            but it doesn't take lists into account
+
+        Parameters
+        ----------
+        user_config : Dict
+            User configuration (~/.samrc) found at user's home directory
+        project_config : Dict
+            Project configuration (.samrc) found at current directory
+
+        Returns
+        -------
+        Dict
+            Merged configuration
+        """
+        # Recursively override User config with Project config
+        for key in user_config:
+            if key in project_config:
+                # If both keys are the same, let's check whether they have nested keys
+                if isinstance(user_config[key], dict) and isinstance(project_config[key], dict):
+                    self.merge_config(user_config[key], project_config[key])
+                else:
+                    user_config[key] = project_config[key]
+                    LOG.debug("Overriding User's key %s with Project's specific value %s.", key, project_config[key])
+
+        # Project may have unique config we need to copy over too
+        # so that we can have user+project config available as one
+        for key in project_config:
+            if key not in user_config:
+                user_config[key] = project_config[key]
+
+        return user_config
+
+    def __has_user_config(self):
```

Comment:
`__has_user_config` and `__has_project_config` are the same methods but do the operation on different files. These can be one method that accept some path to check.

Reply:
Addressed